Sources/MessagingPush/RichPush/RichPushHttpClient.swift

```diff
@@ -0,0 +1,192 @@
+import CioInternalCommon
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+// sourcery: InjectRegisterShared = "HttpClient"
+public class RichPushHttpClient: HttpClient {
```

Comment:
Class is very similar to the `HttPClient` implementation in the `Tracking` module, but stripped down to only contain things needed and decoupled from any `Queue` logic.

Reply:
Instead of introducing a new `HttpClient` implementation inside of the push module, could we re-use the `HttpClient` inside of the Common module? 

I think that adding this file (that looks identical to the one in Common) to the push module would introduce technical debt to the project and I would like to avoid that if possible.

Reply:
`CIOHttpClient` the implementation HttpClient in the `common` would get deleted, since its not being used anymore and was reliant on CIO api/headers etc.

Reply:
I understand that the current `CIOHttpClient` is specific to the Track API. That client is going to go away since we will no longer call the Track API. 

Because the common module contains general classes that can be used by any module in the SDK, it makes the most sense to me for a general `CIOCDPHttpClient` to be contained inside of the common module.

Reply:
`HttpClient` lives in common, if we were to add `CIOCDPHttpClient` in the common as well, we are going to make a decision to bind `common` module with `datapipeline` which doesn't make much sense, since we have multiple modules `push/in-app` and more in future that might just be feature modules independent of datapipeline. 

For example, Gist is using only its own http client for gist services, it doesn't want to utilize `CIOCDPHttpClient`. 

Additionally, `CIOCDPHttpClient` requires, `writeKey` and we won't have that in `common` module.